subprojects/core/src/integTest/groovy/org/gradle/caching/configuration/internal/BuildCacheConfigurationIntegrationTest.groovy

```diff
@@ -47,7 +47,7 @@ class BuildCacheConfigurationIntegrationTest extends AbstractIntegrationSpec {
             gradle.startParameter.buildCacheEnabled = true
             buildCache {
                 local {
-                    directory = '$cacheDir'
+                    directory = file('$cacheDir')
```

Comment:
:x: We should change the definition of `cacheDir` to be a `File` and make these all to:

```suggestion
                    directory = cacheDir
```